### ISSUE TITLE
Why not update the official jQPlot distribution?

### ISSUE THREAD

**Issue by ppritcha** [svandecappelle/jQPlot#51]
Hi guys,  I'm with the real jQPlot.  It's interesting to see that some people are working on updating jQPlot, but it would be better if you would work with me on the official distribution rather than this effort, which is not authorized.  I could use the help and input and I'll facilitate incorporating improvements into the official distibution.  Meanwhile, you may want to clarify that this is not actually jQPlot by using a different name, so as to avoid confusion.

Please contact me, I'll look forward to it.

**Comment by RubenOlsen** [comment]
Hi Paul,

As you may have seen on the mailing list both Johan and I would like to contribute to the original jQPlot. We are really on a roll right now and highly motivated to move jQPlot forward. 

I'd like to still take responsibility for the end user documentation - I have a lot of ideas.

With regards to the documentation - I still think that using GitHub Pages is a OK way to go, but if you have a better idea I am all ears as long as I can do my part to create the best End User documentation on any Open Source project available :-)

**Comment by johanbove** [comment]
Hi Paul, thanks again for reaching out to us. :smile:

I would like to confirm again here that I would still like to contribute to jQplot in any way as long as it stays in the MIT license and the new Github project is part of a jQplot GitHub organization. I believe an organization is necessary since we're now with at least four people investing time and effort on the project.

In all good spirits, what I would like to also have us clarify is your message about the usage of the name "jQplot" in this library and your mention that Steeve's fork is "not authorized"; As far as the MIT license concerns there is no "ownership" of the original open-source library.

Do you, as one of the original contributors of jQplot, have any commercial plans with the name and the software or have plans to change to a different, more restrictive software license?

**Comment by svandecappelle** [comment]
Hi Paul,
I would like to clarify another point.

I think the MIT licence is really important for us. And we want to keep this licence for our project.
About the name of the project, i would like to keep the same name project in terms of the old project seems to not be maintained anymore (There is only an increment of the version number). If maintainers doesn't want to let us keep the same name, we'll change it. 

The actual licence terms on the jqplot project I forked is MIT. So I don't think that my actual repository is a 'not authorized' fork. Could you precise in wich terms do you think it ?

**Comment by RubenOlsen** [comment]
This is really a long rant. Read at your own peril.

# On open source licensing

On the original jQPlot web page (http://www.jqplot.com/info.php) it clearly states that the library is dual licensed. Both the MIT and the GPL v2 clearly states that anyone is allowed to copy the original code. 

**Just to be completely clear: I am not a lawyer**, so what I am about to say is based on working and contributing to open source since 1998.

The good think about having a dual license on a project is that GPL purists can have it their way, and that commercial entities, like my clients, can use and contribute to excellent software. The MIT license even grants an entity to completely engulf an open source project as long as they no claim to have written the software in the first place.

Given that the project is dual licensed, anyone can fork it, (and give it a new name if they so desire). They must however, keep Chris as the original copyright holder. 

It is also important to understand why people fork open source projects: Some time it is because they do not agree with the current maintainers. This does happen quite often.

Other times (like in  our case) it is because a project seems to be abandoned and the only name available in the documentation does not answer any request. This is why an account of contributors is very important in open source (old Open Source projects before the advent of public version control always had a file named CONTRIBUTORS.txt). 

Then we have the cases where and commercial entity add their stash on top of some software or libraries, but clearly does not want to alienate the project contributors. This is how RedHat / CentOS is organized. 

Another example is the split in the SIP Express Router (SER) project. One commercial branch (SER) and one open source called OpenSER. This break was not as cosy as the RedHat / CentOS break. The OpenSER people later renamed their project to Kamaillo to avoid any conflict of interest with the SER people.


# On naming forks of open source projects

Not to be difficult in any way, but pointing out the obvious once and for all: The name jqPlot is owned by Chris Leonello (derivates like jQPlot is really the same name). When forking a project in the realm of bringing it forward because the old project seems to be abandoned - one can for practical purposes keep the name and clearly state that this is a continuation of the old project (this is clearly stated on this GitHub Project). With all due respect, when nothing did happen on BitBucket since 2013, things did appear to be abandoned. 

If this was any other break away from the original jqPlot, the name would probably also have been changed (ref RedHat vs CentOS above).

*The continuation in using the name jQPlot was done in good faith*. Do not shoot us for trying to do good tings :-)


# On new and old names for jQPlot.

The only person who in fact can give away the name is Chris Leonello. No one else.  I does not matter how close you are to Cris, even if you are married to Chris you cannot give away the name. You cannot even legally defend his rights if he does not concur for you to do so as long as the software is licensed as GPL v2 or MIT.

Given that Chris is the original copyright holder, he is the only person who can change the license to a more restrictive one. What he, or anyone else, cannot do, it stop people from forking his his code prior to instigating a more restrictive license.

When this is said: I am still not a lawyer.


# On my personal motivation to contribute to jQPlot

I have been using open source for quite a number of years. Probably more years than most developers living today.  What I see lacking in most open source projects are really good documentation. The not so nice answer to questions about basic usage is very often a variant of *Read the f' source, and do not bother us, we are busy building the software*.

Personally I do not like that attitude, so I try to do my best to crate better documentation. 

I have been using jQPlot since 2009 for several projects. I really like the library - and I once again had the chance to use it in a new project. Naturally I looked for updated source files. Noticed the number of pull requests, the number of issues and the date of the last release. Not very encouraging. Started looking at other solutions like 3D.js - was a bit over the top for my use. Then found this GitHub project. 

Since jQPlot has given me so much in the past, and that I really like the library, made me want to contribute on a fresh fork. I do not have time to do real coding these days, but I am very good at documentation when people leave me in peace to work and organize ;-)

I really do not care where I contribute as long as the organization is transparent and is using a good open source license.


# Where to go next

Based on discussion on the mailing list, and on GitHub my summary is as follows:

* The 3 contributors in the GitHub repo whould like to continue working on the code.
* The 3 contributors have no problem moving their effort to another repository/organization.
* What kind of license is attached to the source is important.

Basically this should be smooth sailing to the next step: 

1. Creating a new GitHub account for the jQPlot while adhering to the current licenses.
2. Add the 3 maintainers of this GitHub repo as maintainer for the new GH repo.
3. Merging the changes that has been done in this repo with the last known good version from BitBucket. As far as I understand, this was already done by Steeve and nothing has been added to the BitBucket repo after Steeve did his fork.

**Comment by svandecappelle** [comment]
All read ! :smile: 

Answer it: `As far as I understand, this was already done by Steeve and nothing has been added to the BitBucket repo after Steeve did his fork.`

@RubenOlsen This is true ! I took the last `master commit` as the fork start.

**Comment by ppritcha** [comment]
Just to clarify...jqPlot is not undergoing a license change.  It will continue under the same license. I'm in the process of setting up the new GitHub account and look forward to working with you.

**Comment by RubenOlsen** [comment]
@ppritcha - I have been asked to be added as a commiter on the jqPlot/jqPlot repo, but nothing has happened. Is there anything I can do to help?
